### ISSUE TITLE
Cookie parsing discards valid cookies

### ISSUE THREAD

**Issue by Qthunder** [http4s/http4s#5877]
When parsing a raw header with multiple cookies, a a failure is returned if any cookie can not be parse. 
So

`Headers(Header.Raw(ci"Cookie", """key1=value1; key2={"aField":{}}"""))).cookies`

will return a parse failure - because the second cookie is invalid, containing non-escaped `"` character - rather than a list of one Cookie (the second).

Perhaps, it should try and salvage valid cookies, by first splitting up the raw header by semi-colons

Gitter discussion:
https://gitter.im/http4s/http4s?at=61e16a3e526fb77b317c2735

**Comment by bplommer** [comment]
I agree. Note: the gitter channel is moribund, most dicussion is on Typelevel discord (https://discord.gg/ZWmFFV73X8)

**Comment by rossabaker** [comment]
This is an interesting idea.  We already have `Headers#getWithWarnings` that returns an `Ior`.  This works great when there are multiple headers of the same name, and we can only parse some of them.

This is the same problem, but the header uses the `;`-delimiter.  (Which is actually quite complicated!  See `org.http4s.internal.parsing.Rfc7230.headerRep1`.)  To make this work generally, what we'd want for recurring headers is to split the semi-colons, and then construct an `Ior` from parsing the components.  Then `getWithWarnings` could work like it does for multiple fields.  This would be slower, because the parsers aren't geared to return an Ior, so I think we'd need to feed it multiple inputs. 🤔

**Comment by Ben-Woolley** [comment]
Chrome is sending cookies from the domain + subdomain in web requests (e.g. example.com + my.example.com). We happen to be the subdomain and there is an invalid cookie from example.com that we have no control over.
